blog/2025-05-23-distr-docs/index.mdx

```diff
@@ -0,0 +1,213 @@
+by external libraries that are integrated.
+
+| Feature                          | Docusaurus                  | Starlight |
+| -------------------------------- | --------------------------- | --------- |
+| Sitemap generation               | ✅                          | ✅        |
+| Metadata & Open Graph            | ✅                          | ✅        |
+| Broken Link Checker              | ✅                          | ✅        |
+| Image Optimization & compression | ⚠️ (hard to get it working) | ✅        |
+
+It should be noted that most features for Starlight are available via plugins.
+
+:::tip
+It is generally recommended to run your documentation through Google's [PageSpeed Insights Test](https://pagespeed.web.dev/)
+:::
+
+### Docusaurus vs. Starlight: Developer experience
+
+A framework for writing documentation should not hold you back from writing documentation quickly:
+
+- It should not be a lot of effort to keep up to date.
+- It should be fast, blazing fast.
+- It should have great integrations, documentation, and community support.
+- It should feel modern and fun to use.
+
+#### Maintenance work
+
+Docusaurus' hard dependency on React introduces a natural update hurdle.
+Although React, Docusaurus, and Infima are all Open Source projects from Meta (formerly Facebook), they are developed by different teams.
+Not only does the Docusaurus team test compatibility with the latest React version, but also all plugins need to ensure compatibility.
+Only after the core team and all plugins support the latest version of React can you safely upgrade your Docusaurus version.
+
+Starlight, on the other hand, only depends on Astro, which is the same team building Starlight, which doesn't seem to require the same amount of effort to keep up to date.
+It also has fewer moving parts.
+
+#### Speed
+
+To be completely honest: The reason I looked for another documentation framework was that I was fed up with Docusaurus taking forever to start the initial development server.
+With the latest Node version (24), Docusaurus version (3.7), and React version (19), it took a little over 5 seconds to start the development server.
+Although this number doesn't seem to be that high, it was a lot higher with previous versions.
+I remember waiting over 20 seconds for the development server to start, which _really_ frustrated me.
+
+Starlight, on the other hand, starts the development server in less than half the time Docusaurus takes.
+The biggest difference is that the Astro compiler [`github.com/withastro/compiler`](https://github.com/withastro/compiler) is written in Go instead of using JavaScript to build the website bundle.
+
+#### Integrations
+
+The only integration that is still missing and not available for Starlight is a Mermaid integration.
+Although Mermaid support ([#1259](https://github.com/withastro/starlight/discussions/1259)) is the second most requested feature after multiple sidebars
```

Comment:
Isn't it a very bad thing if you have to rely on first-party support for some feature in a web-framework? What am I missing?